internal/collector/instance.go

```diff
@@ -43,10 +44,13 @@ func AddToPod(
 	inInstanceConfigMap *corev1.ConfigMap,
 	outPod *corev1.PodSpec,
 	volumeMounts []corev1.VolumeMount,
-	sqlQueryPassword string,
+	sqlQueryPassword,
+	dir string,
 	includeLogrotate bool,
 ) {
-	if !(feature.Enabled(ctx, feature.OpenTelemetryLogs) || feature.Enabled(ctx, feature.OpenTelemetryMetrics)) {
+	if spec == nil ||
```

Comment:
So we want to require the instrumentation spec, then?

Reply:
I think so, just like you said at another point: this is all feature gated, but turning on a feature gate doesn't necessarily mean you want X feature for all your clusters.